db_operations/db.js

```diff
@@ -431,56 +442,155 @@ const registerPass = (email, api_key, lotLocation, gpsLocation, price, notes, ca
         _.forEach(results.records, (record) => {
-          passArray.push(record._fields[0].properties);
+          var passObject = new Object();
+          passObject.avgRating = parseFloat(record.get('totalStars')) / parseFloat(record.get('numberOfRatings'));
+          passObject.gpsLocation = record.get('gpsLocation');
+          passObject.notes = record.get('notes');
+          passObject.forSale = record.get('forSale');
+          passObject.price = record.get('price');
+          passObject.lotLocation = record.get('lotLocation');
+          passObject.passId = record.get('passId');
+          passObject.ownerEmail = record.get('ownerEmail');
+          passObject.ownerCount = record.get('ownerCount');
+
+          passArray.push(passObject);
         });
-        return callback(true, {success:passArray});
+        return callback(true, {success: passArray});
       })
       .catch((e) => {
+        session.close();
         console.log(e);
         return callback(null, {error:'Something went wrong..'});
       })
   } else {
     session
-      .run('MATCH (user:User {api_access_key: {api_key} }), (pass:Pass {forSale: {forSale}}) ' +
-          'WHERE pass.lotLocation CONTAINS {lotLocation} RETURN pass', {
+      .run('MATCH (y:User {api_access_key: { api_key }}) MATCH (user:User), (pass:Pass {forSale: true})'+
+            'WHERE user.email=pass.ownerEmail AND EXISTS(user.totalStars) AND pass.lotLocation CONTAINS {lotLocation}'+
+            'RETURN pass.gpsLocation AS gpsLocation, user.totalStars AS totalStars, user.numberOfRatings AS numberOfRatings,' +
+            'pass.notes AS notes, pass.forSale AS forSale, pass.price AS price, toFloat(pass.ownerCount) AS ownerCount, pass.lotLocation AS lotLocation,'+
+            'pass.id AS passId, pass.ownerEmail AS ownerEmail, y.email AS api_email', {
             forSale: true,
             api_key: api_key,
             lotLocation: keyword
         })
         .then((results) => {
+          session.close();
           if(_.isEmpty(results.records)){
             return callback(false, {error: 'No active users in this lot or no lot match, try again later!'});
           }
           // stores
           let passArray = new Array();
           _.forEach(results.records, (record) => {
-            passArray.push(record._fields[0].properties);
+            var passObject = new Object();
+            passObject.avgRating = parseFloat(record.get('totalStars')) / parseFloat(record.get('numberOfRatings'));
+            passObject.gpsLocation = record.get('gpsLocation');
+            passObject.notes = record.get('notes');
+            passObject.forSale = record.get('forSale');
+            passObject.price = record.get('price');
+            passObject.lotLocation = record.get('lotLocation');
+            passObject.passId = record.get('passId');
+            passObject.ownerEmail = record.get('ownerEmail');
+            passObject.ownerCount = record.get('ownerCount');
```

Comment:
The front end doesn't have a need for 'ownerCount' value at this moment. would you be able to comment this out forthe time being.